vm/src/hint_processor/hint_processor_definition.rs

```diff
@@ -55,6 +51,9 @@ pub trait HintProcessor {
     }
 }
 
+pub trait HintProcessor: HintProcessorLogic + ResourceTracker {}
```

Comment:
How about renaming `HintProcessor` -> `HypervisorContext` and `HintProcessorLogic` -> `HintProcessor`?
(Hypervisor: something "[that creates and runs virtual machines](https://en.wikipedia.org/wiki/Hypervisor)". Seems fitting)

Reply:
Or even just `Hypervisor`...?